### DISCUSSION TITLE
openssl 3 errors causing dropped connections

### DISCUSSION THREAD

**Discussion by davidkbowers** [openssl/openssl#25646] [General Discussion]
We have 4 windows 2016 servers that are exactly the same, hardware and software.  
These servers are running a homegrown message queue.  Each server maintains 2000 connections to offsite servers.

Recently we upgraded our software from openssl 1.1 to openssl 3.0.8.  3  times now in the past 2 weeks on only one of these servers  we have seen it slowly droppimng those connections.  This happens only one server, the other 3 havent had this issue.  Connections continually drop until we restart the application.

When we look at our application logs we see that the server has a forcible disconnect message.  When we look at the disconnect clients we see the following errors from openssl:

10/09 05:19:30|SSL3 alert read:fatal:internal error, File:TaCFdcSSLSocket.cpp, Ln:548
10/09 05:19:30|SSL_connect:Error in error, File:TaCFdcSSLSocket.cpp, Ln:553
10/09 05:19:30|Unknown error, RemoteHost: svrp000ae4aa StatusCode = 183, File:TaCFdcSSLSocket.cpp, Ln:1174
10/09 05:19:30| 1 SSL_connect failed with error Remotehost: svrp000ae4aa.
60020000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:ssl\record\rec_layer_s3.c:1605:SSL alert number 80
, File:TaCFdcSSLSocket.cpp, Ln:1191
10/09 05:19:30|The TLS/SSL connect failed. ErrorCode = 1, File:TaCFdcSSLSocket.cpp, Ln:330

I don't know if anyone can shed some light on this but any suggestions/help is appreciated.

**Comment by nhorman** [comment]
Really not possible to definitively say whats going on from the errors above, but given that you're getting an internal error is looks like you might be getting an invalid record type during the handshake with those clients that are getting dropped.  Would suggest, if you can reproduce reliably, I'd record a tcpdump of the connection and take a look at what the handshake looks like.  It might be that you're reusing a TCP port, or some such, and the server sees you're sending handshakes when it (the server) thinks you should already have an established session
